tasks/Task.cpp

```diff
@@ -28,27 +30,255 @@ bool Task::configureHook()
 {
     if (! TaskBase::configureHook())
         return false;
+
+    // Initialize camera parameters.
+    // The library desconsider the tangencial values, that's why only the
+    // radial values are informed. 
+    frame_helper::CameraCalibration cal = _camera_parameters.get();
+    cam_cal = vpCameraParameters(cal.fx, cal.fy,cal.cx,cal.cy);
+
+    // Set the position of the square target in a frame which origin is
+    // centered in the middle of the square
+    double L = _marker_size.get()/2; // 2L is the square size
+    point[0].setWorldCoordinates(-L,  L, 0);
+    point[1].setWorldCoordinates( L,  L, 0);
+    point[2].setWorldCoordinates( L, -L, 0);
+    point[3].setWorldCoordinates(-L, -L, 0);
+
+    //Center of the target (reference point)
+    P.setWorldCoordinates(0, 0, 0);
+
+    // Define the task
+    task.setServo(vpServo::EYEINHAND_L_cVe_eJe) ;
+    task.setInteractionMatrixType(vpServo::CURRENT, vpServo::PSEUDO_INVERSE);
+
+    switch(_architecture.get().mode)
+    {
+        case IBVS:
+            task.addFeature(p,pd) ;
+            break;
+        case PBVS:
+            task.addFeature(t);
+            task.addFeature(tu);
+            break;
+        case HVS:
+            task.addFeature(p,pd) ;
+            task.addFeature(depth) ;
+            task.addFeature(tu) ;
+    }
+
+    // Set the Jacobian (expressed in the end-effector frame)
+    eJe.eye(6,6);
+    //eJe[3][3] = 0;
+    //eJe[5][3] = 1;
+    task.set_eJe(eJe);
+
+    // Set the pose transformation between the camera and the robot frame
+    Eigen::Affine3d camera2body;
+    _camera2body.get(base::Time::now(), camera2body);
+
+    //Computes the Velocity Twist Matrix from the spatial transformation between the camera and the robot
+    //translation
+    vpTranslationVector cto(camera2body.translation()[0],
+                            camera2body.translation()[1],
+                            camera2body.translation()[2]);
+    //rotation
+    base::Quaterniond q(camera2body.rotation());
+    vpRotationMatrix cRo(vpQuaternionVector(q.x(), q.y(), q.z(), q.w()));
+
+    //build the homogeneous matrix
+    cMe.buildFrom(cto, cRo); 
+    //build and set the velocity twist matrix
+    cVe.buildFrom(cMe);
+    task.set_cVe(cVe);
+
     return true;
 }
+
 bool Task::startHook()
 {
     if (! TaskBase::startHook())
         return false;
     return true;
 }
+
 void Task::updateHook()
 {
     TaskBase::updateHook();
+
+    //read input ports
+    std::vector<base::Vector2d> corners;
+    base::LinearAngular6DCommand setpoint;
+    bool no_corners = (_marker_corners.read(corners) == RTT::NoData);
+    bool no_setpoint = (_cmd_in.read(setpoint) == RTT::NoData);
+
+    //set the state CONTROLLING only if there is setpoint and detected corners
+    if (no_corners) 
+    {
+        state(WAITING_CORNERS);
+    }
+    else if (no_setpoint)
+    {
+        state(MISSING_SETPOINT);
+    }
+    else
+    {
+        state(CONTROLLING);
```

Comment:
Done. Now I handle the states first and the execute the code if the state is CONTROLLING.